smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaLatestCommit.java

```diff
@@ -18,13 +23,33 @@
 public class KafkaLatestCommit implements KafkaCommitHandler {
 
     private final KafkaConsumer<?, ?> consumer;
+    private final Map<TopicPartition, Long> offsets = new HashMap<>();
 
-    public KafkaLatestCommit(KafkaConsumer<?, ?> consumer) {
-        this.consumer = consumer;
+    public KafkaLatestCommit(io.vertx.mutiny.kafka.client.consumer.KafkaConsumer<?, ?> consumer) {
+        this.consumer = (KafkaConsumer<?, ?>) consumer.getDelegate();
     }
 
     @Override
     public <K, V> CompletionStage<Void> handle(IncomingKafkaRecord<K, V> record) {
-        return consumer.commit().subscribeAsCompletionStage();
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        Map<TopicPartition, OffsetAndMetadata> map = new HashMap<>();
+        TopicPartition key = new TopicPartition(record.getTopic(), record.getPartition());
+        Long last = offsets.get(key);
+        // Verify that the latest committed offset before this one.
+        if (last == null || last < record.getOffset() + 1) {
+            offsets.put(key, record.getOffset() + 1);
```

Comment:
**offsets** isn't threadsafe. The user might call the **Message#ack** method inside  a worker thread. It's the reason why on the Throttled strategy we run the handle code inside the context's event thread.

Reply:
Fixed.